internal/testutil/testutil.go

```diff
@@ -94,6 +96,61 @@ func AssertPkgEqual(t *testing.T, sourceDir, destDir string) bool {
 	return assert.Empty(t, diff.List())
 }
 
+// KptfileAwarePkgEqual compares two packages (including any subpackages)
+// and has special handling of Kptfiles to handle fields that contain
+// values which cannot easily be specified in the golden package.
+func KptfileAwarePkgEqual(t *testing.T, pkg1, pkg2 string) bool {
+	diff, err := Diff(pkg1, pkg2)
+	if !assert.NoError(t, err) {
+		return false
+	}
+
+	for _, s := range diff.List() {
+		if !strings.HasSuffix(s, kptfile.KptFileName) {
+			continue
+		}
+
+		pkg1Path := filepath.Join(pkg1, s)
+		pkg1KfExists := kptfileExists(t, pkg1Path)
+
+		pkg2Path := filepath.Join(pkg2, s)
+		pkg2KfExists := kptfileExists(t, pkg2Path)
+
+		if !pkg1KfExists || !pkg2KfExists {
+			continue
+		}
+
+		// Read the Kptfiles and set the Commit field to an empty
+		// string before we compare.
+		pkg1kf, _ := kptfileutil.ReadFile(filepath.Dir(pkg1Path))
```

Comment:
Why is the err being ignored here?

Reply:
I skipped this when I was writing it and forgot to add the error checks. I have updated it.